src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs

```diff
@@ -315,71 +286,95 @@ private string ReadString(int offset, int len)
 
             SeekTo(offset);
 
-            byte[] buffer = GetBuffer(len);
-            if (Stream.Read(buffer, 0, len) != len)
-                return null;
-
-            for (int i = 0; i < len; i++)
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(len);
+            try
             {
-                if (buffer[i] == 0)
+                if (Stream.Read(buffer, 0, len) != len)
+                    return null;
+
+                for (int i = 0; i < len; i++)
                 {
-                    len = i;
-                    break;
+                    if (buffer[i] == 0)
+                    {
+                        len = i;
+                        break;
+                    }
                 }
```

Comment:
Nit:
```C#
int nullTerm = Array.IndexOf(buffer, 0);
if (nullTerm != -1)
    len = nullTerm;
```

Reply:
It may need to be

```csharp
int nullTerm = Array.IndexOf(buffer, 0, 0, len);
```

if the `len` doesn't consider the null terminator.

https://github.com/leculver/clrmd/blob/24aa33e7ee970be03051fcbc6b567cd399341191/src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs#L266-L268

Reply:
> if the len doesn't consider the null terminator

It should go from 0 to Len, but I don't know what you mean about considering a null terminator. We want to search until we either hit a null or reach len.

Reply:
`ArrayPool.Rent` may return a larger array than requested (`len`).

Reply:
> if the `len` doesn't consider the null terminator.

That is... the null terminator is not included in `len` so not read by `Stream.Read`.

`ArrayPool.Rent` may return a larger array than requested (`len`).

Reply:
> if the `len` doesn't consider the null terminator.

That is... the null terminator is not included in `len` thus not read by `Stream.Read`.

`ArrayPool.Rent` may return a larger array than requested (`len`).

Reply:
> may return a larger array than requested (len)

Yes. Ok, I think we're just talking past each other.